### PR TITLE
Update documentation on excluding an auto-configuration to recommend exclude on SpringBootApplication

### PR DESCRIPTION
**Pull request by protyay**
Updated Spring Boot Doc Sec-3.4.2 to encourage use of SpringBootApplication annotation 
@wilkinsona . Please review and merge

### CODE REVIEW

File: spring-boot-project/spring-boot-docs/src/docs/asciidoc/using-spring-boot.adoc [hunk 1]
```diff
@@ -404,12 +404,12 @@ If you find that specific auto-configuration classes that you do not want are be
...
 
 	@Configuration(proxyBeanMethods = false)
-	@EnableAutoConfiguration(exclude={DataSourceAutoConfiguration.class})
+	@SpringBootApplication(exclude={DataSourceAutoConfiguration.class})
 	public class MyConfiguration {
 	}
 ----
 
-If the class is not on the classpath, you can use the `excludeName` attribute of the annotation and specify the fully qualified name instead.
+If you choose to use `@EnableAutoConfiguration` rather than `@SpringBootApplication`, please note that it also has an `exlcude` attribute that can be used. If the class is not on the classpath, you can use the `excludeName` attribute of the annotation and specify the fully qualified name instead.
```
Comment (scottfrederick) [inline spring-boot-project/spring-boot-docs/src/docs/asciidoc/using-spring-boot.adoc]: `exlcude` is misspelled - should be `exclude`

Comment (protyay) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/using-spring-boot.adoc]: Thank you @scottfrederick .
Updated in the latest commit

**Comment by snicoll** [comment]
@protyay thank you for making your first contribution to Spring Boot. Given that `@SpringBootApplication` is added on an application (and is a `@Configuration` already), I've polished the sample [a little bit more](https://github.com/spring-projects/spring-boot/commit/650e55c24200df696e326545a3740fbb5b07e14b).

**Comment by protyay** [comment]
@snicoll Thanks.

**Pull request closed without merging**